tests/test_eoi.py

```diff
@@ -0,0 +1,325 @@
+            TEST_SCHEMA_1,
+            Eoi_Decision.FAIL,
+            [],
+        ),
+    ],
+)
+def test_schema_parsing(answers: dict, schema, exp_result, exp_caveats):
+    # Create a copy of the test form and update with supplied answers for this scenario
+    input_forms = deepcopy(TEST_EOI_FORMS_1)
+    for answer in answers.items():
+        for form in input_forms:
+            for question in form["questions"]:
+                for x in [
+                    field for field in question["fields"] if field["key"] == answer[0]
+                ]:
+                    x["answer"] = answer[1]
+
+    # get a result
+    result = evaluate_eoi_response(schema, input_forms)
+
+    # confirm result is as expected
+    assert result["decision"] == exp_result
+    assert result["caveats"] == exp_caveats
+
+
+TEST_OPERATOR_CONDITIONS = [
+    {
+        "operator": "<=",
+        "compareValue": 4,
+        "result": Eoi_Decision.PASS,
+        "caveat": None,
+    },
+    {
+        "operator": ">=",
+        "compareValue": 5,
+        "result": Eoi_Decision.PASS_WITH_CAVEATS,
+        "caveat": "A caveat: more than 5",
+    },
+    {
+        "operator": "==",
+        "compareValue": 7,
+        "result": Eoi_Decision.PASS_WITH_CAVEATS,
+        "caveat": "A caveat: equals 7",
+    },
+    {
+        "operator": ">",
+        "compareValue": 10,
+        "result": Eoi_Decision.FAIL,
+        "caveat": None,
+    },
+    {
+        "operator": "<",
+        "compareValue": 0,
+        "result": Eoi_Decision.PASS,
+        "caveat": None,
+    },
+]
+
+
+@pytest.mark.parametrize(
+    "answer,exp_decision,exp_caveats",
+    [
+        (2, Eoi_Decision.PASS, []),
+        (-1, Eoi_Decision.PASS, []),
+        (5, Eoi_Decision.PASS_WITH_CAVEATS, ["A caveat: more than 5"]),
+        (
+            7,
+            Eoi_Decision.PASS_WITH_CAVEATS,
+            ["A caveat: more than 5", "A caveat: equals 7"],
+        ),
+        (12, Eoi_Decision.FAIL, []),
+    ],
+)
+def test_evaluate_operators(answer, exp_decision, exp_caveats):
+    result = _evaluate_with_supplied_operators(TEST_OPERATOR_CONDITIONS, answer)
+    assert result[0] == exp_decision
+    assert result[1] == exp_caveats
+
+
+@pytest.mark.parametrize(
+    "operator,exp_exception",
+    [
+        ("<", False),
+        (">", False),
+        ("<=", False),
+        (">=", False),
+        ("==", False),
+        ("<X", True),
+        ("", True),
+        ("randomsomethingelse", True),
+        ("*", True),
+        ("and", True),
+        (None, True),
+    ],
+)
+def test_operator_validation(operator, exp_exception):
+    condition = {
+        "operator": operator,
+        "compareValue": 4,
+        "result": Eoi_Decision.PASS,
+        "caveat": None,
+    }
+    if exp_exception:
+        with pytest.raises(ValueError):
+            result = _evaluate_with_supplied_operators([condition], 1)
+    else:
+        _evaluate_with_supplied_operators([condition], 1)
```

Comment:
Perhaps this could be two distinct tests to avoid the boolean trap anti-pattern?  One for non-exception and one for exceptions?